Add docs for jslib http-instrumentation-pyroscope (#1554)
grafana/k6-docs: docs/sources/next/javascript-api/jslib/http-instrumentation-pyroscope/client.md

---
title: 'Client'

> description: 'Client is a HTTP client attaching tracing information to its requests.'

Author: No - just missed this when I copied the whole thing from the tracing docs.
```suggestion
description: 'Client is a HTTP client attaching baggage headers to its requests.'
```

weight: 02
---

# Client

> `Client` is an HTTP client constructor that attaches baggage header to its requests. Use it to include a context in HTTP requests so that  [Grafana pyroscope](https://grafana.com/oss/pyroscope/) can incorporate their results.

Reviewer: Should baggage header be singular or plural? 
```suggestion
`Client` is an HTTP client constructor that attaches baggage header to its requests.
Use the client to include a context in HTTP requests so that [Grafana pyroscope](https://grafana.com/oss/pyroscope/) can incorporate their results.
```

The `Client` class acts as a drop-in replacement for the standard `http` module and attaches a [baggage header](https://www.w3.org/TR/baggage/) to the request. For details about propagation, refer to [About baggage header](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/http-instrumentation-pyroscope#about-baggage-header).

The `Client` constructor accepts a function taking the method, body and params of each request and returning a map of headers to be added to the request. By default it appends baggage header with info necessary for Grafana Cloud Profiling to integrate with Grafana K6 Ccloud.

For details about propagation, refer to [About baggage header](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/http-instrumentation-pyroscope#about-baggage-header).

## Example

> This example demonstrates how to instantiate a client and use it to instrument HTTP calls. It also illustrates how you can use it alongside the standard `http` module to perform non-instrumented HTTP calls.

Reviewer: The second sentence "It" has an antecedent of HTTP calls. Should this be example and not the call? However, the next "it" (use it alongside) isn't clear. Is this referring to the client or the example? Please feel free to edit this suggestion as needed. 
```suggestion
This example demonstrates how to instantiate a client and use it to instrument HTTP calls. The example also illustrates how you can use the client alongside the standard `http` module to perform non-instrumented HTTP calls.
```

{{< code >}}

```javascript
import { check } from 'k6';
import pyroscope from 'https://jslib.k6.io/http-instrumentation-pyroscope/1.0.0/index.js';
import http from 'k6/http';

> // Explicitly instantiating a pyroscope client allows to distinguish

Reviewer: Capitalize Pyroscope since it's a product name
```suggestion
// Explicitly instantiating a Pyroscope client allows to distinguish
```

// instrumented from non-instrumented HTTP calls, by keeping APIs separate.
const instrumentedHTTP = new pyroscope.Client();

const testData = { name: 'Bert' };

export default () => {
  // Using the pyroscope client instance, HTTP calls will have
  // baggage header added.
  let res = instrumentedHTTP.request('GET', 'http://httpbin.org/get', null, {
    headers: {
      'X-Example-Header': 'instrumented/request',
    },
  });

  // The client offers more flexibility over
  // the `instrumentHTTP` function, as it leaves the

>   // imported standard http module untouched. Thus,

Reviewer: ```suggestion
  // imported standard `http` module untouched. Thus,
```
